secretmanager/api/SecretManagerTest/SecretsFixture.cs

```diff
@@ -0,0 +1,133 @@
+// Copyright(c) 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Xunit;
+
+using Google.Api.Gax.ResourceNames;
+using Google.Cloud.SecretManager.V1Beta1;
+using Google.Protobuf;
+
+namespace GoogleCloudSamples
+{
+    /// <summary>
+    /// Fixture for secrets used in all tests.
+    /// <summary>
+    public class SecretsFixture : IDisposable
+    {
+        private static readonly string s_projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+        private static readonly SecretManagerServiceClient s_client = SecretManagerServiceClient.Create();
+
+        public Secret Secret { get; private set; }
+        public Secret SecretToDelete { get; private set; }
+        public Secret SecretWithVersions { get; private set; }
+        public SecretName SecretToCreateName { get; private set; }
+        public SecretVersion SecretVersion { get; private set; }
+        public SecretVersion SecretVersionToDestroy { get; private set; }
+        public SecretVersion SecretVersionToDisable { get; private set; }
+        public SecretVersion SecretVersionToEnable { get; private set; }
+
+        public SecretsFixture()
+        {
+            if (String.IsNullOrEmpty(s_projectId))
+            {
+                throw new Exception("missing GOOGLE_PROJECT_ID");
+            }
+
+            Secret = CreateSecret();
```

Comment:
If quotas are an issue for managing secrets in the fixture, we can add a tiny sleep after each create/delete/modify operation. But I'm happy for that to be done in a separate PR if we have trouble.